ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CheckDetailsPage.tsx

```diff
@@ -132,6 +174,10 @@ function CheckDetails() {
                     profileName={profileName}
                     tableState={tableState}
                     getSortParams={getSortParams}
+                    searchFilterConfig={searchFilterConfig}
+                    searchFilter={searchFilter}
+                    onSearch={onSearch}
+                    onCheckStatusSelect={onCheckStatusSelect}
```

Comment:
I'm not a fan of having too many props for a component. I believe at this point, we could break this component into more sub-components. This would also divide up responsibilities. It's better to limit the number of responsibilities a component has.

I can either do that here or in a future PR

Reply:
Let me know if any of these sound good or if you have any other suggestions (or combinations of suggestions).

Reply:
I go back and forth on if I think the toolbar belongs in the `Table` component as well. I'd need to spend more time thinking about the 2 approaches you laid out.

Another consideration, could you just pass down `searchFilter` and `setSearchFilter` to the table component? To me it feels like the `onSearch` and `onCheckStatusSelect` logic belongs in `Table`. I'd prefer we keep as much table logic out of the `Page` component as possible, though some logic may needs to remain there due to the URL aspect.

Reply:
+1 for option 2 or something like it. We have a similar pattern in VM 2.0, where the `<Toolbar />` encapsulates the search component, filter chips, and additional filters (severity, status), while the `<Table />` is its own component. The `<Table />` is mostly a dumb component, and receives data, table state, options, etc and the `<Toolbar />` receives `searchFilter` and `setSearchFilter` from the Page component above.

I've definitely been leaning towards keeping everything effectful as high as possible, usually at the Page level if we can, and then passing the state setters down to a callback-named prop ala `onSearch` like @bradr5 mentioned. Although there haven't been any issues that I know of having the various `useURLThing` hooks at multiple levels of the component tree, IMO it is cleaner and easier to work with the closer they are kept to the root. It also makes testing easier, as we don't have to worry about mocking stateful logic that is kept inside the components, as we can just pass whatever we need to `onSearch` during the test.

Reply:
You guys mentioned some good points. I think we could bring up some in our next discussion:

1. Does the `Toolbar` belong in the `Table` or should it be separate? 🤔 I feel like it should be separate. Dave mentioned we separate them in VM 2.0. Brad goes back and forth. It seems like we might be leaning more towards separating them.
2. Should the search state and functions live closer to where they are used, or stay in the `Page` component (root)? 🤔 Dave mentions it is easier, cleaner, and more testing-friendly to work with when they are closer to the root. Brad mentions that we should keep it closer to where they are used and not "pollute" the `Page` component with state and functions it doesn't need to know. I understand both takes but I'm leaning towards keeping it where it is. I'll add the point that passing the `searchFilter` and `setSearch` would also abstract the usage of the state and setter, which might not be a good thing. I think I prefer defining how something should be used (with limits) and giving child components the ability to use them.